modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/listener/DocumentRouteOrphanedListener.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.platform.routing.core.listener;
+
+import static org.nuxeo.ecm.platform.routing.api.DocumentRoutingConstants.ATTACHED_DOCUMENTS_PROPERTY_NAME;
+import static org.nuxeo.ecm.platform.routing.api.DocumentRoutingConstants.DOCUMENT_ROUTE_DOCUMENT_TYPE;
+import static org.nuxeo.ecm.platform.routing.api.DocumentRoutingConstants.ROUTING_TASK_DOC_TYPE;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.PostCommitEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.routing.api.DocumentRoute;
+
+/**
+ * Listener that deletes orphan DocumentRoutes.
+ *
+ * @since 11.2
+ */
+public class DocumentRouteOrphanedListener implements PostCommitEventListener {
+
+    protected static final String QUERY_GET_RELATED_DOCUMENT_ROUTES = "SELECT * FROM " + DOCUMENT_ROUTE_DOCUMENT_TYPE
+            + " WHERE " + ATTACHED_DOCUMENTS_PROPERTY_NAME + " = '%s' ORDER BY dc:created";
```

Comment:
Why the order?

Reply:
Just saw @ataillefer comment... I'm still not sure we need to add a sort here...

Reply:
I thought if you delete a parent workflow you don't need to check the subworkflows, but now that I think about it, but even if it was to work we would run into `DocumentNotFoundException`.

Reply:
In fact, the sorting clause I was talking about was added for the UI, see [NXP-11738](https://jira.nuxeo.com/browse/NXP-11738) and https://github.com/nuxeo-archives/nuxeo-platform-document-routing/commit/bcfa6992c0932e7f307ed6b13acaf06d046553f6, so I think it's indeed not relevant here.